psycopg_pool/psycopg_pool/null_pool.py

```diff
@@ -40,7 +41,55 @@ def _maybe_grow_pool(self) -> None:
         pass
 
 
-class NullConnectionPool(_BaseNullConnectionPool, ConnectionPool):
+class NullConnectionPool(_BaseNullConnectionPool, ConnectionPool[Row]):
+    @overload
+    def __init__(
+        self: "NullConnectionPool[Row]",
+        conninfo: str = ...,
+        *,
+        open: bool = ...,
+        connection_class: Type[Connection[Any]] = ...,
+        configure: Optional[Callable[[Connection[Any]], None]] = ...,
+        reset: Optional[Callable[[Connection[Any]], None]] = ...,
+        kwargs: Optional[Dict[str, Any]] = ...,
+        row_factory: RowFactory[Row],
+        min_size: int = ...,
+        max_size: Optional[int] = ...,
+        name: Optional[str] = ...,
+        timeout: float = ...,
+        max_waiting: int = ...,
+        max_lifetime: float = ...,
+        max_idle: float = ...,
+        reconnect_timeout: float = ...,
+        reconnect_failed: Optional[ConnectFailedCB] = ...,
+        num_workers: int = ...,
+    ):
+        ...
+
+    @overload
+    def __init__(
+        self: "NullConnectionPool[Any]",
```

Comment:
This actually returns a `NullConnectionPool[TupleRow]`. In my merged branch I have used this definition.